### PR TITLE
Rebalance flintlock weapons [RDY]

### PR DESCRIPTION
**Pull request by mugling**
Flintlocks are strictly inferior to cartridge firearms as evidenced by the fact they have no remaining military uses. Prior to this PR this was also true in-game with flintlocks having no effective niche and therefore being effectively unused.

After this PR flintlock weapons remain inaccurate and slow to reload reload and become even slower to aim. They are also further reduced in range compared to cartridge firearms. They are moderately loud and cannot be silenced as they do not accept gunmods (other than shoulder straps).

Their ammo usage is a major advantage requiring neither tools nor primers to craft. It is however only half as efficient when using gunpowder compared to cartridges although it can also use black powder as an alternative. Similar to `shot` it retains interchangeable ranged and `SHOT` types.

Consistent with it's historical use the flintlock rifle becomes a capable melee weapon gaining `DURABLE_MELEE` and  a bayonet. This helps balance it's otherwise severe limitations as a firearm. It gains considerable volume (and some additional) weight.

The flintlock carbine represents a more portable version which also excludes the bayonet. It has reduced range and accuracy compared to the full sized rifle but is somewhat faster to reload. A skilled user has a small chance of getting off a second shot distance and cover permitting.

The flintlock pistol and pepperbox have terrible range. They retain some usefulness when loaded with `paper shot` as a backup weapon in close-quarters.

All of the flintlock weapons will gain an increased chance to misfire when I overhaul the weapon reliability and if anyone can point me in the direction of how to add smoke fields that could be a nice thematic extension

### CODE REVIEW

**Comment by kd8qdz** [comment]
If you are going to make these changes to reflect true smoothbore flintlocks, you should probably add cap muskets (civil war era) as it seems to me that the current "flintlock" is trying to be both.

**Comment by ejseto** [comment]
Map field trails (lightning, plasma, etc.) are added in map::shoot(). There's already a smoke trail flag ("TRAIL") for rocket-type weapons, though that probably isn't what you had in mind since it follows the projectile all the way. You can also just use map::add_field() to add fd_smoke if you know where you want it (e.g. on or in front of the player).

**Comment by mugling** [comment]
> If you are going to make these changes to reflect true smoothbore flintlocks, you should probably add cap muskets (civil war era) as it seems to me that the current "flintlock" is trying to be both.

These changes are aimed at giving the existing items a plausible niche. I'd strongly prefer to rework existing items to include novel and interesting mechanics than add more guns with little differentiating features.

We do have some cap and ball firearms that need some attention if you have any specific suggestions?

> Why would a flint lock be any slower than any other similar length long arm?

Modern weapons typically have shorter barrels and better sights.

**Comment by mugling** [comment]
@ejseto the question then is how much smoke is reasonable to add

**Comment by ejseto** [comment]
That's kinda not what you said, though I suppose it was very slightly ambiguous.

**Comment by Shadefang** [comment]
From what I can tell, bayonets have essentially always been removable.

**Comment by Rivet-the-Zombie** [comment]
>the flintlock rifle becomes a capable melee weapon gaining DURABLE_MELEE and a bayonet

Why not leave this option open to the player in case they don't want a bayonet?

File: data/json/recipes/recipe_ammo.json [hunk 1]
```diff
@@ -3928,22 +3928,19 @@
     "time": 20000,
     "reversible": false,
     "autolearn": true,
-    "book_learn": [[ "manual_rifle", 2 ] , [ "manual_pistol", 2 ] , [ "recipe_bullets", 2 ]],
-    "qualities":[
-      {"id":"CUT","level":1,"amount":1}
-    ],
+    "qualities": [],
```
Comment (Coolthulhu) [inline data/json/recipes/recipe_ammo.json]: Do empty quality arrays need to exist?

Comment (mugling) [inline data/json/recipes/recipe_ammo.json]: Last time I checked yes. Should they need to exist, no.

**Comment by mugling** [comment]
> the flintlock rifle becomes a capable melee weapon gaining DURABLE_MELEE and a bayonet
> 
> Why not leave this option open to the player in case they don't want a bayonet?

The current gunmod system doesn't provide an easy way of then stopping addition of an underslung grenade launcher. TBH a similar problem exists with the shoulder strap.

Unless the current situation is especially problematic I'd like to tackle both when I do some more work on the gunmod code (including trying to close @Coolthulhu's outstanding reported issues)

**Comment by mugling** [comment]
4ad26ab adds `MUZZLE_SMOKE` which is mostly thematic for the flintlock rifle, carbine and pistol but can actually be a hazard when using the pepperbox in enclosed spaces

**Comment by Coolthulhu** [comment]
At least one of the mods complains about the ammo id change during `./cataclysm --check-mods`.

**Comment by mugling** [comment]
I'm building that branch now. Do you happen to know which mod as a quick check with `grep` wasn't especially informative

**Comment by Coolthulhu** [comment]
It may be the vehicle guns one, it has a gattling flintlock or something like this.

**Comment by mugling** [comment]
Indeed, fixed in #15531

**Pull request merged**